extras/kms/kms.go

```diff
@@ -289,55 +289,205 @@ func (k *Kms) CreateKeys(ctx context.Context, scopeId string, purposes []KeyPurp
 		opts.withRandomReader = rand.Reader
 	}
 
+	r, w, localTx, err := k.txFromOpts(ctx, opt...)
+	if err != nil {
+		return fmt.Errorf("%s: %w", op, err)
+	}
+
+	if _, err := createKeysTx(ctx, r, w, rootWrapper, opts.withRandomReader, scopeId, purposes...); err != nil {
+		if localTx != nil {
+			if rollBackErr := localTx.Rollback(ctx); rollBackErr != nil {
+				err = multierror.Append(err, rollBackErr)
+			}
+		}
+		return fmt.Errorf("%s: %w", op, err)
+	}
+	if localTx != nil {
+		if err := localTx.Commit(ctx); err != nil {
+			return fmt.Errorf("%s: unable to commit transaction: %w", op, err)
+		}
+	}
+
+	return nil
+}
```

Comment:
This seems unrelated to the addition of key rotation. Is there some context around this change? The commit that adds this also introduces the `RotateKeys` function, but its not clear how this all relates.

Reply:
this change is a refactor to use `k.txFromOpts(...)` this is a shared func between `CreateKeys` and `RotateKeys`

Reply:
Ah, I think I see it now. `txFromOpts` extracts functionality that used to be directly in `CreateKeys` so that it can then be used by both `CreateKeys` and `RotateKeys`.